src/net_processing.cpp

```diff
@@ -3305,6 +3314,43 @@ bool PeerLogicValidation::SendMessages(CNode* pto, std::atomic<bool>& interruptM
     return true;
 }
 
+void CheckLocalServices(ServiceFlags& nLocalServices) {
+    if (fPruneMode) {
+        LOCK(cs_main);
+
+        // make sure NODE_NETWORK is disabled
+        nLocalServices = ServiceFlags(nLocalServices & ~NODE_NETWORK);
+
+        // don't signal NODE_NETWORK_LIMITED during initial block download
+        if (IsInitialBlockDownload()) {
```

Comment:
Hmm, this feels pretty strange to me. Service bits indicate a commitment to provide certain services, not your current state. More importantly, IBD is always neccessarily fuzzy - we can't know how "in sync" we are.